CraftsmenLtd/BloodConnect
# I-#47: Adding branching related guidelines to doc site

Fixes #47


## What kind of change does this PR introduce (check at least one)


## How would you describe the changes in a changelog (1-2 lines)?
Adding branch management related guidelines to our doc site.

Provide a brief overview of the changes introduced in this pull request.
Added a `rst` file to `/docs/development` named `BranchManagement.rst`, that contains some guidelines on how branches are managed in this repo, including branch naming convention.

## How to test

Build the doc site locally and verify that:
- A link to branch management is visible in `/development/index.html#guidelines`, below `Ticket Lifecycle` page.
- The contents of the page express the intent properly. Please comment if further explanation is needed.

## Checklist


Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.
